From 2b7f26c58a967b35dc094cb285a9cb926c76a881 Mon Sep 17 00:00:00 2001
From: Seungha Yang <seungha@centricular.com>
Date: Wed, 2 Dec 2020 19:46:51 +0900
Subject: [PATCH] proxysink: Always try re-sending stored sticky events per
 upstream sticky event

There might be a sequence that
- Got stream-start/caps/segment events
- Got flush events
- And then buffers with a new segment event

In above case, stream-start and caps event might not be reached to
peer proxysrc.
---
 gst/proxy/gstproxysink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/proxy/gstproxysink.c b/gst/proxy/gstproxysink.c
index f44f8ec46d..309c006730 100644
--- a/gst/proxy/gstproxysink.c
+++ b/gst/proxy/gstproxysink.c
@@ -211,7 +211,7 @@ gst_proxy_sink_sink_event (GstPad * pad, GstObject * parent, GstEvent * event)
     GstPad *srcpad;
     srcpad = gst_proxy_src_get_internal_srcpad (src);
 
-    if (sticky && self->pending_sticky_events) {
+    if (sticky || self->pending_sticky_events) {
       CopyStickyEventsData data = { srcpad, GST_FLOW_OK };
 
       gst_pad_sticky_events_foreach (pad, copy_sticky_events, &data);
-- 
GitLab

